wgsl/index.bs

```diff
@@ -9249,6 +9249,51 @@ Any node which is not visited by these reachability analyses can be proven to be
 
 Note: The bottom-up analysis is still required, as it lets us know what edges to add to the graphs when encountering calls.
 
+### Examples ### {#uniformity-examples}
+
+The graphs in the subsequent example use the following conventions for nodes:
+* Rectangles represent value nodes.
+* Rounded  rectangles represent control flow nodes.
+
+#### Invalid `textureSample` Function Call #### {#uniformity-example1}
+
+This example shows an invalid use of a [[#texturesample|textureSample]]
+built-in function call.
+The function call is made within an if statement whose condition depends on a
+non-uniform value (i.e. the built-in value `position`).
+The invalid dependency chain is highlighted in red.
+
+<div class='example wgsl' heading='WGSL'>
+  <xmp highlight='rust'>
+    @group(0) @binding(0) var t : texture_2d<f32>;
+    @group(0) @binding(1) var s : sampler;
+
+    @fragment
+    fn main(@builtin(position) pos : vec4<f32>) {
+      if (pos.x < 0.5) {
+        _ = textureSample(t, s, pos.xy);
+      }
+    }
+  </xmp>
+</div>
+
+<figure>
+  <figcaption>Uniformity graph</figcaption>
+  <object type="image/svg+xml" data="img/uniformity_1.mmd.svg"></object>
```

Comment:
The diagram does not show up in the preview.
The bikeshed generation uses a tarball that contains all the sources.  Maybe you have to add the diagram subdirectory to the tarball?
Also the index.pre.html target should depend on `diagrams`

See https://github.com/gpuweb/gpuweb/blob/ed4030a32ee652413921e90602db4bb1ce2b0abb/wgsl/Makefile#L12

Reply:
The bikeshed tarball only needs to have the bikeshed sources in it so it can generate index.html. The images/svgs don't get embedded in index.html. So `make diagrams` only needs to be run if the diagrams need to be regenerated.